app/assets/javascripts/modules/m3_viewer.js

```diff
@@ -77,5 +87,30 @@ export default class M3Viewer {
       miradorDownloadDialogPlugin,
       miradorDownloadPlugin,
     ]);
+    if (data.search) {
+      const state = mirador.store.getState();
+      const windowId = Mirador.selectors.getWindowIds(state)[0];
+      const companionWindowId = Mirador.selectors.getCompanionWindowIdsForPosition(state, { windowId: windowId, position: 'left' })[0];
+      let searchService = Mirador.selectors.getManifestSearchService(state, { windowId: windowId });
+      // Waits for 2 seconds for searchService to appear (manifest loaded and parsed)
+      let counter = 0;
+      const checkForSearchService = setInterval(() => {
```

Comment:
Since state is async here.. we have to keep checking if the manifest comes in and is parsed, before we can get the search service. Perhaps we can provide a better API in Mirador? But this works for now.

Reply:
I'm wondering about possibly adding the ability to initiate a mirador viewer/window w/ a search initialized (since Content Search is in core).

UV has this feature and I think it would be beneficial to support something similar.  It would also allow us to avoid maintaining some of this code that is highly coupled to the Mirador API.

Reply:
I would be interested in exploring this in Mirador.  This is a feature UV has, and I think it would be beneficial for us to have it to.

Reply:
It seems like if this was in the Mirador API we could avoid any concern about having to set/clear intervals... which would be nice.

Reply:
Totally agree! I could pop off the last commit if we want to get the other stuff in. We could then recircle to the mirador API stuff to handle search?

Reply:
I'll create a quick ticket over in Mirador.

Reply:
K. I've done that. Draft PR of search stuff here: https://github.com/sul-dlss/sul-embed/pull/1027